lib/events/test/suite.go

```diff
@@ -107,6 +107,11 @@ func (s *EventsSuite) EventPagination(t *testing.T) {
 	var err error
 	var checkpoint string
 
+	// For eventually consistent queries
+	if s.QueryDelay != 0 {
```

Comment:
it''s needed because of eventually consistent s3 based athena solution. I can rework RetryStaticFor used below to include it, but didn't want to modify test suite too much. Let me know what do you think? 

same applied to all other queryDelays sleeps.

Reply:
We can consider wrapping (only for integration test purpose) the AthenaAuditLogger with a new  internal object  object that will be responsible for handling eventually consistency dependency between  SearchEvents and EmitAuditEvent by delaying SearchEvents call by `QueryDelay` from the last `EmitAuditEvent` call so we will not need hardcode the if s.QueryDelay != 0 { time.Sleep()} calls for backend types.

Reply:
On the other hand both dynamo and firestore use that param: https://github.com/gravitational/teleport/blob/d6af208bf646245215bb3dd8b2299c0d7497909e/lib/events/dynamoevents/dynamoevents_test.go#L80
it was just often used before first query, not after every emitting.

Reply:
I have used Marek suggestion to wrap athena inside new object only for tests and handle there eventual consistency: https://github.com/gravitational/teleport/pull/25248/commits/a8162c719fdc523a9ce7215c3b54e74edb713b96